src/middleware/log.ml

```diff
@@ -109,21 +109,20 @@ let reporter () =
 
       (* Format the current local time. For the millisecond fraction, be careful
          of rounding 999.5+ to 1000 on output. *)
-      let time =
-        let open Unix in
+      let time ~now =
```

Comment:
Nit, but given that `~now` is passed to `reporter`, is there any reason to pass it to `time`, and turn `time` into a function, here?

Reply:
No reason, however it should be better to add a unit argument may be to show the **real** time when we make the log's line (and do not pre-compute `time`)? It's minor and it can be a part of another PR (unrelated to MirageOS).

Reply:
> No reason, however it should be better to add a unit argument may be to show the **real** time when we make the log's line (and do not pre-compute `time`)? It's minor and it can be a part of another PR (unrelated to MirageOS).

I don't understand the intent behind this comment. On the one hand, getting the actual time is already suspended because `now` is a function of `unit`, and we can just move the application of `now` somewhere else.

On the other hand, this comment sounds like you are suggesting we should delay evaluating the time. But don't we want to advance evaluating the time instead, to the soonest point when the log was actually executed? It's already quite late as it is, in the reporter. I think that's probably fine, it's okay to accept a small amount of imprecision.

Reply:
> On the other hand, this comment sounds like you are suggesting we should delay evaluating the time. But don't we want to advance evaluating the time instead, to the soonest point when the log was actually executed? It's already quite late as it is, in the reporter. I think that's probably fine, it's okay to accept a small amount of imprecision.

Yes, I'm not really clear with my english, sorry. But this what I tried to say. So you can do what you want, just to notice you this specific point 🙂 .